feat(range-input): initial design guidance (#31)
vmware-clarity/clarity-guidance: 1023-range-input-design-guidance.md

---
category: component design guidance
state: draft
created: 2024-10-18
updated: 2024-10-22
---

# Range Input Design Guidance

> [Range Input](https://clarity.design/documentation/range) consists of a slider that can be moved to represent some value between the minimum and maximum values. It allows the user to select a value when the precise value chosen is not considered necessary.

Reviewer: ```suggestion
A [range Input](https://clarity.design/documentation/range) consists of a slider that can be moved to represent some value between the minimum and maximum values. It allows the user to select a value when the precise value chosen is not considered necessary.
```

> A [range Input](https://clarity.design/documentation/range) consists of a slider that can be moved to represent some value between the minimum and maximum values. It allows the user to select a value when the precise value chosen is not considered necessary.

Reviewer 2: ```suggestion
A [range input](https://clarity.design/documentation/range) features a slider that allows users to select a value within a defined minimum and maximum range. It’s ideal for scenarios where a precise value isn’t required.
```

## Guidance

> - **Must** include clear labels indicating the purpose of the slider.

Reviewer 2: ```suggestion
- Design teams **must** include clear labels indicating the purpose of the slider to ensure users understand what value they are selecting.
```

- **Should** use a label above or beside the slider, rather than below it, in order to remain visible while the user is selecting a value.
- **Should** use [radio buttons](https://clarity.design/documentation/radio) or a [select](https://clarity.design/documentation/select) when only a few distinct choices are available or a number [input](https://clarity.design/documentation/input) when precision is key.

> - **Should** use appropriate [range input states](https://clarity.design/documentation/range#states).

Reviewer 2: ```suggestion
- Design teams **should** use appropriate [range input states](https://clarity.design/documentation/range#states) to provide visual feedback and enhance the user experience during interaction.
```

> - **May** use range input component when it is acceptable to have an imprecise input control and the exact value is not important.

Reviewer 2: ```suggestion
- Design teams **may** use the range input component when it's okay to have a less precise input and the exact value doesn’t matter, allowing users to quickly pick a value from a wider range.
```

---
category: component design guidance
state: draft
created: 2024-10-18
updated: 2024-10-21
---

# Range Input Design Guidance

[Range Input](https://clarity.design/documentation/range) consists of a slider that can be moved to represent some value between the minimum and maximum values. It allows the user to select a value when the precise value chosen is not considered necessary.

## Guidance

> - **Must** use range component when it is acceptable to have an imprecise input control and the exact value is not important.

Reviewer: ```suggestion
- **Must** use range input component when it is acceptable to have an imprecise input control and the exact value is not important.
```

- **Must** include clear labels indicating the purpose of the slider.
- **Should** use a label above or beside the slider, rather than below it, in order to remain visible while the user is selecting a value.

> - **Should** use UI components like Radio button or Dropdown, when only a few distinct choices are available or when precision is key.

Reviewer: ```suggestion
- **Should** use [radio buttons](https://clarity.design/documentation/radio) or a [select](https://clarity.design/documentation/select) when only a few distinct choices are available or a number [input](https://clarity.design/documentation/input) when precision is key.
```

Added links. Changed dropdown to select because a dropdown cannot be used as a form control. Added number input as an option when precision is needed. Technically, combobox and datalist are also options in addition to select, but the wording can get long. I am open to suggestions.

> - Use appropriate [states for Range Input](https://clarity.design/documentation/range#states).

Reviewer: ```suggestion
- *Should* use appropriate [range input states](https://clarity.design/documentation/range#states).
```

> 

Reviewer: ```suggestion
```

nit: extra blank line
